src/charts/chart.js

```diff
@@ -281,110 +271,28 @@ export default function Chart (_container) {
       .style("width", `${config.markPanelWidth}px`)
       .style("height", `${config.chartHeight}px`)
       .attr("fill", "transparent")
-
     return this
   }
 
   function buildChart () {
-    scales = components.scale
+    config = transformConfig(cache.config)
```

Comment:
the auto config, data transform and scales are now explicitly happening here rather than as a side-effect on the setters.

Reply:
okay, that makes sense. But is `buildChart` now only being called when the chart first renders and `update` being called on re-renders? If so the config and data wouldn't be updated correct?

Reply:
`buildChart` is called same as before to render all the components. The only difference is that the main chart `update` doesn't happen in `build`, it happens inside `buildChart` so it can use the same config and data that is being transformed at the start of this method.